packages/gatsby/src/schema/infer-graphql-input-fields.js

```diff
@@ -273,6 +273,10 @@ export function inferInputObjectStructureFromNodes({
       const nodeToFind = isArray ? value[0] : value
       const linkedNode = findLinkedNode(nodeToFind)
 
+      if (!linkedNode) {
+        return
```

Comment:
Can you just add some context here in comment - ie. that this just prevents crash and it will fail in `infer-graphql-type.js`, but will provide more context to user?